### PR TITLE
[FE] Fix profile update methods

### LINKED ISSUE #357: When trying to edit the profile, I get a 500
image

### PR DESCRIPTION
**Pull request by Drakulix**
Fixes #357

### CODE REVIEW

**Comment by suzanap** [comment]
you can still not change your password as normal user

File: web/src/components/common/userComponents/ProfileEditor.jsx [hunk 1]
```diff
@@ -173,14 +173,29 @@ export default class ProfileEditor extends Component {
             </div>
           </div>
           <div className="row">
-            <label className="col-2">New Password</label>
+            <label className="col-2">Old password</label>
+            <div className="col-4">
+              <TextField
```
Comment (Drakulix) [inline web/src/components/common/userComponents/ProfileEditor.jsx]: Hide this is current user is admin please.

File: web/src/components/common/userComponents/ProfileEditor.jsx [hunk 2]
```diff
@@ -173,14 +173,29 @@ export default class ProfileEditor extends Component {
...
+                          value={this.state.pw_old}
+                          type="password"
+                          name="pw_old"
+                          hintText="Your old password"
+                          onChange={this.handleInputChange}
+                          errorText ={this.state.pw_new.length === 0 ? "Password can not be empty" :""} />
```
Comment (Drakulix) [inline web/src/components/common/userComponents/ProfileEditor.jsx]: Should this error text not check for `this.state.pw_old`?

File: web/src/components/common/userComponents/ProfileEditor.jsx [hunk 3]
```diff
@@ -202,7 +217,7 @@ export default class ProfileEditor extends Component {
                           type="Submit"
                           label="Change Password"
                           primary={true}
-                          disabled= {(this.state.pw_new !== this.state.pw_new_confirm || this.state.pw_new === "") ? true : false}
+                          disabled= {(this.state.pw_new !== this.state.pw_new_confirm || this.state.pw_new === "" || this.state.pw_old === "") ? true : false}
```
Comment (Drakulix) [inline web/src/components/common/userComponents/ProfileEditor.jsx]: This should also not check for old password if editing user is admin

**Review by Drakulix — COMMENTED** [review]
@mharasic Not complete, but almost there.

File: web/src/components/common/userComponents/ProfileEditor.jsx (hunk 2 above)
Comment (mharasic) [reply web/src/components/common/userComponents/ProfileEditor.jsx]: We dont have our old PW ...

Comment (Drakulix) [reply web/src/components/common/userComponents/ProfileEditor.jsx]: yes, but this is the old pw field and you are checking if `this.state.pw_new.length` is not 0

**Review by mharasic — APPROVED** [review]
Worked so far

**Pull request merged**